packages/dcos-integration-test/extra/test_packaging.py

```diff
@@ -0,0 +1,152 @@
+        packages = sorted(map(lambda id_: tuple(id_.split('--')), package_ids))
+        assert len(packages) == len(active_buildinfo_packages)
+        for package, buildinfo_package in zip(packages, active_buildinfo_packages):
+            if buildinfo_package[1] is None:
+                # No buildinfo for this package, so we can only compare names.
+                assert package[0] == buildinfo_package[0]
+            else:
+                assert package == buildinfo_package
+
+    for node in dcos_api_session.masters + dcos_api_session.all_slaves:
+        package_ids = get_and_validate_package_ids('pkgpanda/repository/', node)
+        active_package_ids = get_and_validate_package_ids('pkgpanda/active/', node)
+
+        assert set(active_package_ids) <= set(package_ids)
+        assert_packages_match_active_buildinfo(active_package_ids)
+
+
+# There is no standardized way of getting package requirements from the package description json,
+# e.g. nodes could be called 'brokers' or 'nodes' or something else. This was created by looking at
+# https://github.com/mesosphere/universe/blob/version-3.x/repo/packages/K/kafka/39/config.json
+# 1.1.9-0.10.0.0
+KAFKA_PACKAGE_REQUIREMENTS = {
+    'number_of_nodes': 3,
+    'node': {
+        'disk': 5000,
+        'mem': 2304,
+        'cpus': 1
+    },
+    'executor': {
+        'disk': 0,
+        'mem': 256,
+        'cpus': 0.5
+    }
+}
+
+
+def get_cluster_resources(dcos_api_session):
+    """Return the mesos state summary
+    """
+    r = dcos_api_session.get('mesos/state-summary')
+    return r.json()
+
+
+def enough_resources_for_package(state_summary, package_requirements):
+    """SANITY CHECK that there are enough unreserved_resources in a state to run a package.
+
+    Args:
+        state_summary: dict
+        package_requirements: dict
+
+    Returns:
+        bool
+
+    Notes:
+        This is a sanity check meant for use with pytest.mark.skipif
+        This only verifies that there were enough resources at the time state_summary was queried
+    """
+    agents = state_summary['slaves']
+    if len(agents) < package_requirements['number_of_nodes']:
+        log.debug('Not enough agents for this package. Need {required}, have {available}'.format(
+            required=package_requirements['number_of_nodes'],
+            available=len(agents)))
+        return False
+    usable_nodes = 0
+    for agent in agents:
+        if agent_has_resources(agent, package_requirements['node']):
+            usable_nodes += 1
+        if usable_nodes == package_requirements['number_of_nodes']:
+            return True
+    log.info('Only {available} usable agents. This package needs {required} with these resources {resources}.'.format(
+        available=usable_nodes,
+        required=package_requirements['number_of_nodes'],
+        resources=package_requirements['node']))
+    return False
+
+
+def agent_has_resources(agent, node_requirements):
```

Comment:
If we go by definition follows use. This function should be _private and should be defined above _enough_resources_for_package.